File: make_fsf/r/count_nifti_volumes.R
# Install the oro.nifti package if not already installed
# install.packages("oro.nifti")

library(oro.nifti)

count_nifti_volumes <- function(nifti_file) {
  tryCatch({
    nii <- readNIfTI(nifti_file)
    num_volumes <- dim(nii$data)[4]  # Assuming 4th dimension represents time points
    return(num_volumes)
  }, error = function(e) {
    message(paste("Error loading or processing", nifti_file, ":", e))
    return(NULL)
  })
}

# Example usage:
nifti_file <- "your_nifti_file.nii.gz"
num_volumes <- count_nifti_volumes(nifti_file)
if (!is.null(num_volumes)) {
  cat(paste("Number of volumes in", nifti_file, ":", num_volumes, "\n"))
}

File: make_fsf/python/count_nifti_volumes.py
import nibabel as nib

def count_nifti_volumes(nifti_file):
    try:
        nifti = nib.load(nifti_file)
        data = nifti.get_fdata()
        num_volumes = data.shape[-1]  # Assuming last dimension represents time points
        return num_volumes
    except Exception as e:
        print(f"Error loading or processing {nifti_file}: {e}")
        return None

# Example usage:
nifti_file = 'your_nifti_file.nii.gz'
num_volumes = count_nifti_volumes(nifti_file)
if num_volumes is not None:
    print(f"Number of volumes in {nifti_file}: {num_volumes}")